### PR TITLE
First run at adding RNR episodes, additional refactoring, and docs

### PR DESCRIPTION
**Pull request by jamonholmgren**
This is a WIP -- I'll explain in more detail later. It probably won't live in the main tab bar...maybe.

image

### CODE REVIEW

File: boilerplate/app/i18n/ar.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
   "demoNavigator": {
     "componentsTab": "عناصر",
     "debugTab": "تصحيح",
+    "listTab": "TODO: List",
```
Comment (frankcalise) [inline boilerplate/app/i18n/ar.json:30]: Should we change this to `MAVERICKTODO` ?

File: boilerplate/app/models/EpisodeStore.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { Instance, SnapshotOut, types } from "mobx-state-tree"
+import { api } from "../services/api"
+import { EpisodeModel } from "./Episode"
+import { withSetPropAction } from "./helpers/with-set-prop-action"
+
+export const EpisodeStoreModel = types
+  .model("EpisodeStore")
+  .props({
+    title: "React Native Radio",
```
Comment (yulolimum) [inline boilerplate/app/models/EpisodeStore.ts:9]: we probably don't need this here

File: boilerplate/app/navigators/DemoNavigator.tsx [hunk 3]
```diff
@@ -54,6 +56,17 @@ export function DemoNavigator() {
         }}
       />
 
+      <Tab.Screen
+        name="DemoList"
+        component={DemoListScreen}
+        options={{
+          tabBarLabel: translate("demoNavigator.listTab"),
+          tabBarIcon: ({ focused }) => (
+            <Icon icon="menu" style={{ tintColor: focused && colors.tint }} />
```
Comment (yulolimum) [inline boilerplate/app/navigators/DemoNavigator.tsx:65]: we probably want a podcast icon from @JenFucci

File: boilerplate/app/screens/DemoListScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import { DemoTabScreenProps } from "../navigators/DemoNavigator"
+import { colors } from "../theme"
+
+export function DemoListScreen(props: DemoTabScreenProps<"DemoList">) {
+  const { episodeStore } = useStores()
+
+  useEffect(() => {
+    // kick off fetching episodes from the API
+    episodeStore.fetchEpisodes()
+  })
```
Comment (yulolimum) [inline boilerplate/app/screens/DemoListScreen.tsx:15]: this should have empty dep array

**Review by yulolimum — APPROVED** [review]
left a couple comments. otherwise this is aight

File: boilerplate/app/i18n/ar.json (hunk 1 above)
Comment (yulolimum) [reply boilerplate/app/i18n/ar.json:30]: ya jamon what the heck?

File: boilerplate/app/models/helpers/with-set-prop-action.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ *    })
+ *    .actions(withSetPropAction)
+ *
+ *   const user = UserModel.create({ name: "Jamon", age: 40 })
+ *
+ *   user.setProp("name", "John") // no type error
+ *   user.setProp("age", 30)      // no type error
+ *   user.setProp("age", "30")    // type error -- must be number
+ */
+export const withSetPropAction = <T extends IStateTreeNode>(mstInstance: T) => ({
```
Comment (mazenchami) [inline boilerplate/app/models/helpers/with-set-prop-action.ts:24]: ok, this is cool 🎉

**Review by mazenchami — APPROVED** [review]

File: boilerplate/app/i18n/ar.json (hunk 1 above)
Comment (jamonholmgren) [reply boilerplate/app/i18n/ar.json:30]: CALLED OUT

File: boilerplate/app/models/EpisodeStore.ts (hunk 2 above)
Comment (jamonholmgren) [reply boilerplate/app/models/EpisodeStore.ts:9]: Probably not, yeah

File: boilerplate/app/navigators/DemoNavigator.tsx (hunk 3 above)
Comment (jamonholmgren) [reply boilerplate/app/navigators/DemoNavigator.tsx:65]: Good call.

File: boilerplate/app/screens/DemoListScreen.tsx (hunk 4 above)
Comment (jamonholmgren) [reply boilerplate/app/screens/DemoListScreen.tsx:15]: Ouch, yes

File: boilerplate/app/navigators/DemoNavigator.tsx (hunk 3 above)
Comment (yulolimum) [reply boilerplate/app/navigators/DemoNavigator.tsx:65]: @jamonholmgren btw, Icon has convenience prop called `color` so you don't have to specify a style.

**Comment by jamonholmgren** [comment]
Latest commit should fix CI.

File: boilerplate/app/navigators/DemoNavigator.tsx (hunk 3 above)
Comment (jamonholmgren) [reply boilerplate/app/navigators/DemoNavigator.tsx:74]: Oops, I missed this @yulolimum , maybe you can fix in your PR

File: boilerplate/app/config/index.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ * This file imports configuration objects from either the config.dev.js file
+ * or the config.prod.js file depending on whether we are in __DEV__ or not.
+ *
+ * Note that we do not gitignore these files. Unlike on web servers, just because
+ * these are not checked into your repo doesn't mean that they are secure.
+ * In fact, you're shipping a JavaScript bundle with every
+ * config variable in plain text. Anyone who downloads your app can easily
+ * extract them.
+ *
+ * If you doubt this, just bundle your app, and then go look at the bundle and
```
Comment (frankcalise) [inline boilerplate/app/config/index.ts:11]: I like this note 👍

**Pull request merged**